public/app/plugins/datasource/elasticsearch/elastic_response.ts

```diff
@@ -451,11 +417,17 @@ export class ElasticResponse {
         this.trimDatapoints(tmpSeriesList, target);
         this.nameSeries(tmpSeriesList, target);
 
+        if (table.rows.length > 0) {
+          dataFrame.push(toDataFrame(table));
+        }
+
         for (let y = 0; y < tmpSeriesList.length; y++) {
           let series = toDataFrame(tmpSeriesList[y]);
 
           // When log results, show aggregations only in graph. Log fields are then going to be shown in table.
-          series = addPreferredVisualisationType(series, 'graph');
+          if (isLogsRequest) {
+            series = addPreferredVisualisationType(series, 'graph');
```

Comment:
Something minor I forgot yesterday - when looking at what this does, I realized that `addPreferredVisualisationType` both mutates `series` **and** returns it. While one could refactor it to be either one **or** the other, I suppose simply inlining this very short function that's only called here would be another option.